components/pdfmerger/components/PDFMerger/PDFMerger.tsx

```diff
@@ -75,27 +71,37 @@ export default function PDFMerger() {
         }
       }
       const dims = image.scaleToFit(pageW, pageH);
-      const page = pdf.addPage([pageW, pageH]);
+      index = getActualIndex(index);
```

Comment:
imo, the design of this function is a bit weird, `getActualIndex` implicitly requires `pdf` which is a state inside the component. I would have thought that this is a stateless function at first, even though it's not.

But I'm not familiar with the overall code to suggest a recommendation.
1. Why does `getActualIndex` require `pdf`? Is it because the page has already been added to `pdf`? If that's so, what if we maintain the page in another state somewhere before adding it to `pdf`?
2. I think we should differentiate the `index` name

Reply:
It requires because the pdf could be null hence we can't calculate the page index. I created this function because adding pages to the pdf is an async process and file A which has an order number less than file B can finish embedding longer than file B, thus file B will be added to the pdf first. It will not matter if it's an images only because the page index for each image is the same as the index in file list. However it kinda gets messy when we have pdfs to merge (because it has > 1 pages to embed from a single file). I actually thought of another solution which to save the return value for each addImage, addPdf etc in a state (?) then add it to the pdf at the end but haven't really tried that and i'm not sure if it's better that way.